src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs

```diff
@@ -36,24 +37,52 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                 var contentItem = token.ToObject<ContentItem>();
                 var modifiedUtc = contentItem.ModifiedUtc;
                 var publishedUtc = contentItem.PublishedUtc;
+                var isPublished = contentItem.Published;
+                var isLatest = contentItem.Latest;
+
                 var existing = await _contentManager.GetVersionAsync(contentItem.ContentItemVersionId);
-                
+
                 if (existing == null)
                 {
                     // Initializes the Id as it could be interpreted as an updated object when added back to YesSql
                     contentItem.Id = 0;
                     await _contentManager.CreateAsync(contentItem);
-                    
+
                     // Overwrite ModifiedUtc & PublishedUtc values that handlers have changes
                     // Should not be necessary if IContentManager had an Import method
                     contentItem.ModifiedUtc = modifiedUtc;
                     contentItem.PublishedUtc = publishedUtc;
-                }
-                else
-                {
-                    // Replaces the id to force the current item to be updated
-                    existing.Id = contentItem.Id;
-                    _session.Save(existing);
+
+                    // Reset published and latest to imported values as CreateAsync sets these values arbitrarily.
+                    contentItem.Published = isPublished;
+                    contentItem.Latest = isLatest;
```

Comment:
Here I'm not sure why you need to reset these values as you don't do a CreateAsync in this "else" statement.

Reply:
so actually the code shown here is hard to read.

I have removed the else statement.

As it does / did nothing except force an existing item that had just been retrieved from the database (and not mutated) to be resaved.

I'm not saying we should necessarily remove it permanently, but unless we do some mutation on existing content (i.e. the same version id) then it makes no sense.

I'm not sure that mutating an existing contentitemversion is the right thing to do either. 
The idea should be (IMHO) that altering a contentitem via a recipe, means creating a new version to do so, i.e. that's what a deployment plan should do, update content that has changed.

Unless we want to support editing a recipe by hand, to mutate existing content without creating a new version id.

Reply:
I agree